server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Plan.hs

```diff
@@ -130,54 +117,83 @@ data CohortVariables
   -- >   }
   -- > }
   --
-  -- might be executed with similar values for @$condition@, such as @{"album_id": {"_eq": "1"}}@
-  -- and @{"album_id": {"_eq": "2"}}@.
+  -- might be executed with similar values for @$condition@, such as @{"album_id":
+  -- {"_eq": "1"}}@ and @{"album_id": {"_eq": "2"}}@.
   --
-  -- Normally, we wouldn’t bother parameterizing over the @1@ and @2@ literals in the resulting
-  -- query because we can’t cache that query plan (since different @$condition@ values could lead to
-  -- different SQL). However, for live queries, we can still take advantage of the similarity
-  -- between the two queries by multiplexing them together, so we replace them with references to
-  -- synthetic variables.
+  -- Normally, we wouldn’t bother parameterizing over the @1@ and @2@ literals in the
+  -- resulting query because we can’t cache that query plan (since different
+  -- @$condition@ values could lead to different SQL). However, for live queries, we
+  -- can still take advantage of the similarity between the two queries by
+  -- multiplexing them together, so we replace them with references to synthetic
+  -- variables.
   } deriving (Show, Eq, Generic)
 instance Hashable CohortVariables
 
+-- | Builds a cohort's variables by only using the session variables that
+-- are required for the subscription
+mkCohortVariables
+  :: Set.HashSet SessVar
+  -> UserVars
+  -> ValidatedQueryVariables
+  -> ValidatedSyntheticVariables
+  -> CohortVariables
+mkCohortVariables requiredSessionVariables sessionVariableValues =
+  CohortVariables requiredSessionVariableValues
+  where
+    requiredSessionVariableValues =
+      mapUserVars
+      (Map.filterWithKey (\k _ -> Set.member k requiredSessionVariables))
+      sessionVariableValues
+
 instance J.ToJSON CohortVariables where
   toJSON (CohortVariables sessionVars queryVars syntheticVars) =
-    J.object ["session" J..= sessionVars, "query" J..= queryVars, "synthetic" J..= syntheticVars]
+    J.object [ "session" J..= sessionVars
+             , "query" J..= queryVars
+             , "synthetic" J..= syntheticVars
+             ]
 
 -- These types exist only to use the Postgres array encoding.
-newtype CohortIdArray = CohortIdArray { unCohortIdArray :: [CohortId] }
+newtype CohortIdArray
+  = CohortIdArray { unCohortIdArray :: [CohortId] }
```

Comment:
Why do this? The original line was under 80 characters!

Reply:
It must have happened because of muscle memory - that's the code style I'm used to (and most of the current code is in this form if I'm not mistaken)!